cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -1324,6 +1361,555 @@ TEST_F(ScalarTemporalTest, TestTemporalDifferenceZoned) {
+  const char* round_15_months =
+      R"(["1970-01-01", "2000-01-01", "1898-10-01", "2033-10-01", "2020-01-01",
+          "2020-01-01", "2020-01-01", "2010-01-01", "2010-01-01", "2010-01-01",
+          "2010-01-01", "2006-04-01", "2006-04-01", "2008-10-01", "2008-10-01",
+          "2012-07-01", null])";
+  const char* round_15_quarters =
+      R"(["1970-01-01", "2000-01-01", "1898-10-01", "2033-10-01", "2018-10-01",
+          "2018-10-01", "2018-10-01", "2011-04-01", "2011-04-01", "2011-04-01",
+          "2011-04-01", "2007-07-01", "2007-07-01", "2007-07-01", "2007-07-01",
+          "2011-04-01", null])";
+  const char* round_15_years =
+      R"(["1965-01-01", "1995-01-01", "1905-01-01", "2040-01-01", "2025-01-01",
+          "2025-01-01", "2025-01-01", "2010-01-01", "2010-01-01", "2010-01-01",
+          "2010-01-01", "2010-01-01", "2010-01-01", "2010-01-01", "2010-01-01",
+          "2010-01-01", null])";
+
+  auto unit = timestamp(TimeUnit::NANO, "UTC");
+  CheckScalarUnary(op, unit, times, unit, round_1_nanoseconds, &round_to_1_nanoseconds);
+  CheckScalarUnary(op, unit, times, unit, round_1_microseconds, &round_to_1_microseconds);
+  CheckScalarUnary(op, unit, times, unit, round_1_milliseconds, &round_to_1_milliseconds);
+  CheckScalarUnary(op, unit, times, unit, round_1_seconds, &round_to_1_seconds);
+  CheckScalarUnary(op, unit, times, unit, round_1_minutes, &round_to_1_minutes);
+  CheckScalarUnary(op, unit, times, unit, round_1_hours, &round_to_1_hours);
+  CheckScalarUnary(op, unit, times, unit, round_1_days, &round_to_1_days);
+  CheckScalarUnary(op, unit, times, unit, round_1_weeks, &round_to_1_weeks);
+  CheckScalarUnary(op, unit, times, unit, round_1_months, &round_to_1_months);
+  CheckScalarUnary(op, unit, times, unit, round_1_quarters, &round_to_1_quarters);
+  CheckScalarUnary(op, unit, times, unit, round_1_years, &round_to_1_years);
+
+  CheckScalarUnary(op, unit, times, unit, round_15_nanoseconds, &round_to_15_nanoseconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_microseconds,
+                   &round_to_15_microseconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_milliseconds,
+                   &round_to_15_milliseconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_seconds, &round_to_15_seconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_minutes, &round_to_15_minutes);
+  CheckScalarUnary(op, unit, times, unit, round_15_hours, &round_to_15_hours);
+  CheckScalarUnary(op, unit, times, unit, round_15_days, &round_to_15_days);
+  CheckScalarUnary(op, unit, times, unit, round_15_weeks, &round_to_15_weeks);
+  CheckScalarUnary(op, unit, times, unit, round_15_months, &round_to_15_months);
+  CheckScalarUnary(op, unit, times, unit, round_15_quarters, &round_to_15_quarters);
+  CheckScalarUnary(op, unit, times, unit, round_15_years, &round_to_15_years);
+
+  RoundTemporalOptions round_to_2_hours = RoundTemporalOptions(2, CalendarUnit::HOUR);
```

Comment:
We need to #ifdef these out on Windows right?

Reply:
I think the whole bottom of the file is already and it might not show in the diff.

Reply:
Oops, I didn't expand it out enough. Thanks.